assert/assertions.go

```diff
@@ -1707,21 +1707,35 @@ func NotRegexp(t TestingT, rx interface{}, str interface{}, msgAndArgs ...interf
 }
 
 // Zero asserts that i is the zero value for its type.
+// If i implements IsZero it is preferred over checking the zero value.
 func Zero(t TestingT, i interface{}, msgAndArgs ...interface{}) bool {
 	if h, ok := t.(tHelper); ok {
 		h.Helper()
 	}
+	if zeroer, ok := i.(interface{ IsZero() bool }); ok {
+		if zeroer.IsZero() {
+			return true
+		}
+		return Fail(t, fmt.Sprintf("Should be zero, but was %v", i), msgAndArgs...)
+	}
 	if i != nil && !reflect.DeepEqual(i, reflect.Zero(reflect.TypeOf(i)).Interface()) {
 		return Fail(t, fmt.Sprintf("Should be zero, but was %v", i), msgAndArgs...)
 	}
 	return true
 }
 
 // NotZero asserts that i is not the zero value for its type.
+// If i implements IsZero it is preferred over checking the zero value.
```

Comment:
What is IsZero? You would need to fully document it as encoding/json does.

Reply:
```suggestion
// If i has a IsZero() bool method it is preferred over checking the zero value.
```